### ISSUE TITLE
[QUESTION] .NET 5 Support

### ISSUE THREAD

**Issue by DaNeubi** [litedb-org/LiteDB#1990]
Hello there,

is there a plan to officially support .NET 5?

Did anyone use this library in .NET 5 already?

Tbh, I would like to contribute, but my C# knowledge isn't that advanced compared to all of the contributors. A label "**good first issue**" would be great imo.

**Comment by AleRoe** [comment]
I've been using LiteDb for months now in a net core environment without any issues.

**Comment by jsavage7** [comment]
I was able to add .net5 support in my fork of litedb. You would need to remove net45 support (you could change it to net472). This is because System.Reflection.TypeExtensions only supports Net 4.6 and up. You would do this in the LiteDB project file. then update the type extension with nuget package.

Changes in project file:

<TargetFrameworks>net472;netstandard1.3;netstandard2.0;net50</TargetFrameworks>


  <PropertyGroup Condition="'$(TargetFramework)' == 'net50'">
    <DefineConstants>HAVE_SHA1_MANAGED;HAVE_APP_DOMAIN;HAVE_PROCESS;HAVE_ENVIRONMENT</DefineConstants>
  </PropertyGroup>


  <PropertyGroup Condition="'$(TargetFramework)' == 'net472'">
    <DefineConstants>HAVE_SHA1_MANAGED;HAVE_APP_DOMAIN;HAVE_PROCESS;HAVE_ENVIRONMENT</DefineConstants>
  </PropertyGroup>

**Comment by JKamsker** [comment]
Yes, and it has been since .NET 5 came out. LiteDB 5.x (current stable: 5.0.21) ships a `netstandard2.0` build, alongside `netstandard1.3` and `net45`. .NET 5 and every later .NET version consume that build directly, so you don't need a `net5.0`-specific target or a fork. Just reference the package:

```
dotnet add package LiteDB
```

The 6.x prereleases also ship a native `net8.0` build next to `netstandard2.0`.

Keep in mind that .NET 5 itself went out of support in May 2022, so we'd recommend .NET 8 or later. #2403 covers the same question for .NET 8.

Closing this as answered — if it doesn't cover your case, comment here and we'll reopen it.
